lib/mongo/grid/stream/read.rb

```diff
@@ -167,7 +167,8 @@ def ensure_readable!
           end
 
           def view
-            @view ||= fs.chunks_collection.find({ :files_id => file_id }, options).read(read_preference).sort(:n => 1)
+            @view ||= (opts = options.merge(read: read_preference) if read_preference
```

Comment:
I think we can just keep all the logic in one place, esp since calling a separate method is less efficient.